[CI] StatelessFileDeletionIT testCommitsOfScrollAreDeletedAfterIndexIsClosedAndOpened failing

**Build Scans:**
- [elasticsearch-intake #41610 / part2](https://gradle-enterprise.elastic.co/s/qzvgvxkftkjck)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:stateless:internalClusterTest" --tests "org.elasticsearch.xpack.stateless.commits.StatelessFileDeletionIT.testCommitsOfScrollAreDeletedAfterIndexIsClosedAndOpened" -Dtests.seed=6B42FB639206898D -Dtests.locale=ps-AF -Dtests.timezone=America/Nome -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.stateless.commits.StatelessFileDeletionIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testCommitsOfScrollAreDeletedAfterIndexIsClosedAndOpened'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: a value greater than <16>
     but: <16> was equal to <16>
```

**Issue Reasons:**
- [main] 1 consecutive failures in test testCommitsOfScrollAreDeletedAfterIndexIsClosedAndOpened
- [main] 1 consecutive failures in step part2
- [main] 1 consecutive failures in pipeline elasticsearch-intake

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -611,9 +611,6 @@ tests:
 - class: org.elasticsearch.xpack.logsdb.LogsdbRestIT
   method: testEsqlRuntimeFields
   issue: https://github.com/elastic/elasticsearch/issues/151547
-- class: org.elasticsearch.xpack.stateless.commits.StatelessFileDeletionIT
-  method: testCommitsOfScrollAreDeletedAfterIndexIsClosedAndOpened
-  issue: https://github.com/elastic/elasticsearch/issues/151558
 - class: org.elasticsearch.xpack.esql.qa.multi_node.GenerativeIT
   method: test {feature:BASELINE}
   issue: https://github.com/elastic/elasticsearch/issues/149746

--- a/x-pack/plugin/stateless/src/internalClusterTest/java/org/elasticsearch/xpack/stateless/commits/StatelessFileDeletionIT.java
+++ b/x-pack/plugin/stateless/src/internalClusterTest/java/org/elasticsearch/xpack/stateless/commits/StatelessFileDeletionIT.java
@@ -924,8 +924,13 @@ public class StatelessFileDeletionIT extends AbstractStatelessPluginIntegTestCas
             var blobsAfterReleasingScroll = listBlobsWithAbsolutePath(shardCommitsContainer);
             assertThat(Sets.intersection(blobsUsedForScroll, blobsAfterReleasingScroll), empty());
         });
-        // responses from newCommitNotification must be received
-        assertThat(countNewCommitNotifications.get(), greaterThan(newCommitNotificationsBeforeReleasingScroll));
+
+        // For most cases we expect at least one additional newCommitNotification response after releasing the scroll.
+        // Except for the close/open case, where notifications induced by the forced flush (in
+        // TransportVerifyShardBeforeCloseAction.executeShardOperation) may race with the search shard closing and may not be counted.
+        if (testCase != TestSearchScrollCase.COMMITS_OF_SCROLL_DELETED_AFTER_INDEX_CLOSED_AND_OPENED) {
+            assertThat(countNewCommitNotifications.get(), greaterThan(newCommitNotificationsBeforeReleasingScroll));
+        }
 
         if (testCase == TestSearchScrollCase.COMMITS_DROPPED_AFTER_SCROLL_CLOSES_AND_INDEXING_INACTIVITY) {
             // The last notification response should no longer contain blobs used for scroll

